unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenController.cs

```diff
@@ -113,14 +116,17 @@ private void TeleportRequested(Vector3 current, Vector3 previous)
                 //Temporarily removing tips until V2
                 //tipsController.StopTips();
                 percentageController.StartLoading(currentDestination);
-
                 view.FadeIn(false, true);
-            }
-            else
-            {
-                //We are going to check if the scene has timeout using the POSITION_SETTLED event.
-                CheckSceneTimeout(currentDestinationCandidate);
-            }
+                StartTimeoutCounter(timeoutCTS.Token);
+            }else if (IsSceneLoaded(currentDestinationCandidate))
+                HandlePlayerLoading();
+        }
+
+
+        private async UniTaskVoid StartTimeoutCounter(CancellationToken ct)
+        {
+            await UniTask.Delay(LOAD_SCENE_TIMEOUT, cancellationToken: ct);
```

Comment:
I would change `LOAD_SCENE_TIMEOUT` to `TimeSpan.FromMilliseconds(LOAD_SCENE_TIMEOUT)` for the sake of readability